datafusion/sqllogictest/test_files/copy.slt

```diff
@@ -251,30 +261,30 @@ query IT
 COPY source_table
 TO 'test_files/scratch/copy/table_with_options/'
 (format parquet,
-compression snappy,
-'compression::col1' 'zstd(5)',
-'compression::col2' snappy,
-max_row_group_size 12345,
-data_pagesize_limit 1234,
-write_batch_size 1234,
-writer_version 2.0,
-dictionary_page_size_limit 123,
-created_by 'DF copy.slt',
-column_index_truncate_length 123,
-data_page_row_count_limit 1234,
-bloom_filter_enabled true,
-'bloom_filter_enabled::col1' false,
-'bloom_filter_fpp::col2' 0.456,
-'bloom_filter_ndv::col2' 456,
-encoding plain,
-'encoding::col1' DELTA_BINARY_PACKED,
-'dictionary_enabled::col2' true,
-dictionary_enabled false,
-statistics_enabled page,
-'statistics_enabled::col2' none,
-max_statistics_size 123,
-bloom_filter_fpp 0.001,
-bloom_filter_ndv 100
+'format.parquet.compression' snappy,
```

Comment:
This is a solid example of how we change the option handling.

Reply:
It would be nice if the config options didn't need to be literal quoted e.g. 'format.parquet.compression'. I think it would be ideal if DFParser were aware of the custom syntax we are creating for options including the dots and double colons, so a user could simply write:

```sql
COPY table
TO 'file.parquet'
(
format.parquet.compression snappy,
format.parquet.compression::col1 zstd(5)
)
```

Related is #9274

Reply:
Sounds good. We can enhance this behaviour after this PR.